presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -130,20 +149,64 @@ private URI buildQuery(URI baseURI, String time, String metricName, String query
         BigDecimal maxQueryRangeDecimal = BigDecimal.valueOf(maxQueryRangeDuration.getSeconds()).add(BigDecimal.valueOf(maxQueryRangeDuration.getNano(), 9));
         BigDecimal queryChunkSizeDecimal = BigDecimal.valueOf(queryChunkSizeDuration.getSeconds()).add(BigDecimal.valueOf(queryChunkSizeDuration.getNano(), 9));
 
-        int numChunks = maxQueryRangeDecimal.divide(queryChunkSizeDecimal).setScale(0, RoundingMode.UP).intValue();
+        int numChunks = maxQueryRangeDecimal.divide(queryChunkSizeDecimal, 0, RoundingMode.UP).intValue();
 
         return Lists.reverse(IntStream.range(0, numChunks).mapToObj(n -> {
-            long endTime = now.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli() -
+            long endTime = upperBound.toInstant().toEpochMilli() -
                     n * queryChunkSizeDuration.toMillis() - n * OFFSET_MILLIS;
             return endTime;
         }).map(endTimeMilli -> decimalSecondString(endTimeMilli)).collect(Collectors.toList()));
     }
 
+    protected static Optional<PrometheusPredicateTimeInfo> determinePredicateTimes(TupleDomain<ColumnHandle> predicate)
+    {
+        Optional<Map<ColumnHandle, Domain>> maybeColumnHandleDomainMap = predicate.getDomains();
+        Optional<Set<ColumnHandle>> maybeKeySet = maybeColumnHandleDomainMap.map(columnHandleDomainMap -> columnHandleDomainMap.keySet());
+        Optional<Set<ColumnHandle>> maybeOnlyPromColHandles = maybeKeySet.map(keySet -> keySet.stream()
+                .filter(columnHandle -> columnHandle instanceof PrometheusColumnHandle).collect(Collectors.toSet()));
+        Optional<Set<ColumnHandle>> maybeOnlyTimeStampColumnHandles = maybeOnlyPromColHandles.map(columnHandles -> columnHandles.stream()
+                .filter(colHandle -> ((PrometheusColumnHandle) colHandle).getColumnType() instanceof TimestampType).collect(Collectors.toSet()));
```

Comment:
I know we don't have any other timestamp columns, but maybe we should be checking the PrometheusCololumnHandle name too?

Reply:
Done in 7203da7b. And turns out all the tests had the column misnamed, so good catch.